controllers/topology/desired_state_test.go

```diff
@@ -419,6 +419,88 @@ func TestComputeCluster(t *testing.T) {
 	g.Expect(obj.Spec.ControlPlaneRef).To(Equal(objToRef(controlPlane)))
 }
 
+func TestComputeMachineDeployment(t *testing.T) {
```

Comment:
Let's add a test case for a machine deployment referencing a class that does not exists